client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -0,0 +1,390 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import isEmpty from 'lodash/lang/isEmpty';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'analytics';
+import acceptDialog from 'lib/accept';
+import DisconnectJetpackDialog from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog';
+import PluginItem from 'my-sites/plugins/plugin-item/plugin-item';
+import PluginsActions from 'lib/plugins/actions';
+import PluginsListHeader from 'my-sites/plugins/plugin-list-header';
+import PluginsLog from 'lib/plugins/log-store';
+import PluginNotices from 'lib/plugins/notices';
+import SectionHeader from 'components/section-header';
+
+export default React.createClass( {
+	displayName: 'PluginsList',
+
+	mixins: [ PluginNotices ],
+
+	propTypes: {
+		plugins: PropTypes.array,
+		header: PropTypes.string,
+		isWpCom: PropTypes.bool,
+		sites: PropTypes.object,
+		selectedSite: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ),
+		pluginUpdateCount: PropTypes.number,
+		isPlaceholder: PropTypes.bool,
+	},
+
+	getInitialState() {
+		return {
+			disconnectJetpackDialog: false,
+			bulkManagement: false,
+			selectedPlugins: {}
+		};
+	},
+
+	componentDidMount() {
+		PluginsLog.on( 'change', this.showDisconnectDialog );
+	},
+
+	componentWillUnmount() {
+		PluginsLog.removeListener( 'change', this.showDisconnectDialog );
+	},
+
+	isSelected( plugin ) {
+		return !! this.state.selectedPlugins[ plugin.slug ];
+	},
+
+	togglePlugin( plugin ) {
+		let selectedPlugins = Object.assign( {}, this.state.selectedPlugins );
+		selectedPlugins[ plugin.slug ] = ! this.state.selectedPlugins[ plugin.slug ];
+		this.setState( { selectedPlugins } );
```

Comment:
There another way to accomplish the toggle in one line of logic. One reason I like this approach is that we can see in a single place the single action that this function is taking. Of course, we use a couple of helper variables to make it clearer, but those values never change and are basically just there for the cleanliness of it.

```js
const { slug } = plugin;
const { selectedPlugins } = this.state;
const oldValue = selectedPlugins[ slug ];

this.setState( { selectedPlugins: { ...selectedPlugins, [ slug ]: ! oldValue } } );
```

I could have used `Object.assign()` here too, but in this case I found the new spread operator more helpful to show what exactly I wanted to do with the existing array - update a particular key with a new (and flipped) value.